### ISSUE TITLE
broken procedure that generates a key during setup

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#1303]
Ok, so there are three issues when creating a key during setup:
 - missing / unclear details of that error message. It should be showing clearly the URL and http code like other API errors do. This will need an appium test as well. The Attester mock will be set to return a 400 on submission with some error message, and the above modal must show that error message (we had a mechanism for this that was supposed to work across modals)
 - the fact that there is an error submitting to attester
 - per Dmitry's account, whenever there isn't the above error, backup doesn't get sent to inbox

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/issues/484#issuecomment-1011990874_

**Comment by Kharchevskyi** [comment]
First issue I faced now is that call to `MailServiceProviderFactory.services` executes not on main thread both for gmail and imap
<img width="400" alt="Screenshot 2022-01-14 at 22 01 39" src="https://user-images.githubusercontent.com/13693255/149577673-72f94f66-bd10-4ba0-a4ca-7f73786a7cbc.png">

Which cause issue with Main thread checker
<img width="400" alt="Screenshot 2022-01-14 at 22 03 07" src="https://user-images.githubusercontent.com/13693255/149577846-64f3c34d-92ef-42ff-ab7b-ee04e8196ece.png">

@sosnovsky probably would need assist with this issue. 
_Will fill a separate issue for that_

**Comment by Kharchevskyi** [comment]
looks like call to attester returns 409 status code
```
[URLSession] URLSession.call status:409 ms:175 PUT https://flowcrypt.com/attester/pub/anton_flowcrypt@outlook.com, headers: [:]
```
just wondering should it be post instead? 
<img width="400" alt="Screenshot 2022-01-14 at 22 19 19" src="https://user-images.githubusercontent.com/13693255/149579803-6f612256-4dc9-46b0-86a2-74c1841a9a0d.png">

if yes, what token to use here for imap? 
<img width="400" alt="Screenshot 2022-01-14 at 22 20 17" src="https://user-images.githubusercontent.com/13693255/149579904-aa84d89f-1c54-4dac-9c0b-49408f3f4b73.png">

**Comment by tomholub** [comment]
Make it a POST, and add id_token to the authorization header.

**Comment by Kharchevskyi** [comment]
@tomholub this is for outlook authentication.
Mailcore uses SASL way of authentication

<img width="346" alt="Screenshot 2022-01-16 at 15 08 07" src="https://user-images.githubusercontent.com/13693255/149661229-fd334f21-7f71-44b6-a162-a4412c00b692.png">

**Comment by tomholub** [comment]
On that case, we need to use a PUT for IMAP. But we have to add a modal with retry/skip option.

**Comment by Kharchevskyi** [comment]
@tomholub do we still need to do smth in scope of this issue after #1311 merged?

**Comment by tomholub** [comment]
@fcvakintos does it behave as expected for you now?
